woocommerce/wc-admin
# SummaryNumber Component: Update to latest hifi design

See #131 - this PR adds a previous value/label to each SummaryNumber, and updates the design to match the updated hifi designs. This also adds in the ability to link off to a report by adding a required `href` prop to the SummaryNumber.

The smaller screen dropdown functionality will come in a later PR, as that will be much more complex.

**Demo on 1440px wide:**

<img width="572" alt="large-screen-one" src="https://user-images.githubusercontent.com/541093/43209619-1972bac6-8ffb-11e8-9ba4-3f02d271ec13.png">
<img width="948" alt="large-screen-two" src="https://user-images.githubusercontent.com/541093/43209620-19837cda-8ffb-11e8-833f-a4cdd5ac5879.png">

**Demo on 1024px wide:**

<img width="629" alt="med-screen-two" src="https://user-images.githubusercontent.com/541093/43209626-1d7f31bc-8ffb-11e8-9961-5865a2b4c12d.png">

**Demo in a card**

<img width="844" alt="screen shot 2018-07-25 at 11 09 03 am" src="https://user-images.githubusercontent.com/541093/43209650-2da87684-8ffb-11e8-9818-273243ad68a2.png">

**Demo of interactive states**

(top left is active/pressed, middle is selected, top right is hovered, bottom left is unselected + focused)
<img width="791" alt="screen shot 2018-07-25 at 1 30 50 pm" src="https://user-images.githubusercontent.com/541093/43217250-1d832e34-900f-11e8-956d-647cd111f901.png">

(right is a selected + focused item)
<img width="790" alt="screen shot 2018-07-25 at 1 31 04 pm" src="https://user-images.githubusercontent.com/541093/43217251-1d8ed928-900f-11e8-8a3b-9445db51efc2.png">

Don't be scared off by the `+676 lines`, 374 of that is creating an extensive demo page with lots of different number combos 😁

To test:

- Look at the example report `/wp-admin/admin.php?page=wc-admin#/analytics/test`
- Check that all SummaryNumbers appear as expected on screens larger than 783px (have not addressed the dropdowns yet)
- Specific "edge cases":
  - The 2nd item in most cases is "selected", the exception is in "Ten or More Data Points" where it's the 5th item to show selection on the second line.
  - The 3rd item in most cases has a delta of 0
  - In "Four Data Points", the 3rd item has no previous value or delta
  - In "Ten or More Data Points", the 8th item has a ridiculously large previous value
- Mouse or keyboard over them to check on the interactive states
- Check the same for the SummaryNumber card on the dashboard
- Try using VoiceOver to interact with the SummaryNumbers – I've called this a navigation menu, since they link off to specific reports/charts. You should be able to target and click on a summary number

**Note** All the summary numbers go to a default link of `/analytics`, because a) I don't know what these should link to, and b) I really didn't want to update every single demo `SummaryNumber` 😆
